source/Nuke.Common/Execution/BuildManager.cs

```diff
@@ -51,11 +52,22 @@ void ExecuteExtension<TExtension>(Action<TExtension> action)
                 Logger.OutputSink = build.OutputSink;
                 Logger.LogLevel = NukeBuild.LogLevel;
 
-                ToolPathResolver.ExecutingAssemblyDirectory = NukeBuild.BuildProjectDirectory != null
-                    ? Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)
-                    : null;
-                ToolPathResolver.NuGetAssetsConfigFile = NukeBuild.BuildProjectDirectory / "obj" / "project.assets.json";
-                ToolPathResolver.NuGetPackagesConfigFile = build.NuGetPackagesConfigFile;
+                if(NukeBuild.BuildProjectDirectory != null)
+                {
+                    var assemblyLocation = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+                    ToolPathResolver.ExecutingAssemblyDirectory = EnvironmentInfo.ValidDirectoryPath(assemblyLocation)
+                        ? assemblyLocation
+                        : Path.GetDirectoryName(Process.GetCurrentProcess().MainModule?.FileName);
+
+                    ToolPathResolver.NuGetAssetsConfigFile = NukeBuild.BuildProjectDirectory / "obj" / "project.assets.json";
+                    ToolPathResolver.NuGetPackagesConfigFile = build.NuGetPackagesConfigFile;
```

Comment:
Previously we tried to assign null to **NukeBuild.BuildProjectDirectory** without checking it's not null.